Caused by: java.lang.ClassCastException: com.trello.lifecycle2.android.lifecycle.AndroidLifecycle_LifecycleAdapter cannot be cast to android.arch.lifecycle.GeneratedAdapter

I am getting this when i call bindToLifecycle now after updating to android arch lifecycle extensions 1.0.0-beta2 and android support library 26.1.0

Caused by: java.lang.ClassCastException: com.trello.lifecycle2.android.lifecycle.AndroidLifecycle_LifecycleAdapter cannot be cast to android.arch.lifecycle.GeneratedAdapter
                                                                  at android.arch.lifecycle.Lifecycling.createGeneratedAdapter(Lifecycling.java:79)
                                                                  at android.arch.lifecycle.Lifecycling.getCallback(Lifecycling.java:62)
                                                                  at android.arch.lifecycle.LifecycleRegistry$ObserverWithState.<init>(LifecycleRegistry.java:319)
                                                                  at android.arch.lifecycle.LifecycleRegistry.addObserver(LifecycleRegistry.java:145)
                                                                  at com.trello.lifecycle2.android.lifecycle.AndroidLifecycle.<init>(AndroidLifecycle.java:36)
                                                                  at com.trello.lifecycle2.android.lifecycle.AndroidLifecycle.createLifecycleProvider(AndroidLifecycle.java:30)
                                                                  at com.trello.rxlifecycle2.android.lifecycle.kotlin.RxlifecycleKt.bindToLifecycle(rxlifecycle.kt:23)

Looking at the code in the lifecycle arch lib.. i believe the AndroidLifecycle class needs to inherit from GenericLifecycleObserver or FullLifecycleObserver now

--- a/.travis.yml
+++ b/.travis.yml
@@ -4,7 +4,7 @@ android:
   components:
     - tools
     - platform-tools
-    - build-tools-26.0.1
+    - build-tools-26.0.2
     - android-26
     - extra-android-m2repository
 

--- a/build.gradle
+++ b/build.gradle
@@ -4,6 +4,7 @@ buildscript {
         verKotlin = '1.1.4-3'
     }
     repositories {
+        google()
         jcenter()
     }
     dependencies {

--- a/build.gradle
+++ b/build.gradle
@@ -29,7 +29,8 @@ ext {
     buildToolsVersion = '26.0.1'
 
     supportLibVersion = '26.0.2'
-    lifecycleVersion = '1.0.0-alpha9'
+    lifecycleVersion = '1.0.0'
+    lifecycleRuntimeVersion = '1.0.3'
 
     sourceCompatibilityVersion = JavaVersion.VERSION_1_7
     targetCompatibilityVersion = JavaVersion.VERSION_1_7
@@ -38,7 +39,7 @@ ext {
     rxJava = 'io.reactivex.rxjava2:rxjava:2.1.3'
     rxAndroid = 'io.reactivex.rxjava2:rxandroid:2.0.1'
     navi = 'com.trello.navi2:navi:2.0'
-    lifecycle = "android.arch.lifecycle:runtime:$lifecycleVersion"
+    lifecycle = "android.arch.lifecycle:runtime:$lifecycleRuntimeVersion"
     lifecycleProcessor = "android.arch.lifecycle:compiler:$lifecycleVersion"
     lifecycleExtensions = "android.arch.lifecycle:extensions:$lifecycleVersion"
     kotlinStdlib = "org.jetbrains.kotlin:kotlin-stdlib:$verKotlin"

--- a/build.gradle
+++ b/build.gradle
@@ -7,7 +7,7 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.3.3'
+        classpath 'com.android.tools.build:gradle:3.0.0'
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$verKotlin"
         classpath "org.jetbrains.kotlin:kotlin-android-extensions:$verKotlin"
         classpath 'com.github.ben-manes:gradle-versions-plugin:0.15.0'
@@ -26,7 +26,7 @@ ext {
     minSdkVersion = 14
     compileSdkVersion = 26
     targetSdkVersion = compileSdkVersion
-    buildToolsVersion = '26.0.1'
+    buildToolsVersion = '26.0.2'
 
     supportLibVersion = '26.0.2'
     lifecycleVersion = '1.0.0'

--- a/rxlifecycle-android-lifecycle/src/test/java/com/trello/rxlifecycle2/android/lifecycle/AndroidLifecycleActivityTest.java
+++ b/rxlifecycle-android-lifecycle/src/test/java/com/trello/rxlifecycle2/android/lifecycle/AndroidLifecycleActivityTest.java
@@ -3,7 +3,6 @@ package com.trello.rxlifecycle2.android.lifecycle;
 import android.arch.lifecycle.Lifecycle;
 import android.arch.lifecycle.LifecycleActivity;
 import android.arch.lifecycle.LifecycleOwner;
-import android.arch.lifecycle.LifecycleRuntimeTrojanProvider;
 import com.trello.lifecycle2.android.lifecycle.AndroidLifecycle;
 import com.trello.rxlifecycle2.LifecycleProvider;
 import io.reactivex.Observable;
@@ -26,8 +25,6 @@ public class AndroidLifecycleActivityTest {
     @Before
     public void setup() {
         observable = PublishSubject.create().hide();
-        // LifecycleRegistryOwner requires a content provider to init correctly.
-        Robolectric.buildContentProvider(LifecycleRuntimeTrojanProvider.class).create();
     }
 
     @Test

--- a/rxlifecycle-android-lifecycle/src/test/java/com/trello/rxlifecycle2/android/lifecycle/AndroidLifecycleFragmentTest.java
+++ b/rxlifecycle-android-lifecycle/src/test/java/com/trello/rxlifecycle2/android/lifecycle/AndroidLifecycleFragmentTest.java
@@ -3,7 +3,6 @@ package com.trello.rxlifecycle2.android.lifecycle;
 import android.arch.lifecycle.Lifecycle;
 import android.arch.lifecycle.LifecycleFragment;
 import android.arch.lifecycle.LifecycleOwner;
-import android.arch.lifecycle.LifecycleRuntimeTrojanProvider;
 import android.support.v4.app.Fragment;
 import android.support.v4.app.FragmentActivity;
 import com.trello.lifecycle2.android.lifecycle.AndroidLifecycle;
@@ -27,8 +26,6 @@ public class AndroidLifecycleFragmentTest {
     @Before
     public void setup() {
         observable = PublishSubject.create().hide();
-        // LifecycleRegistryOwner requires a content provider to init correctly.
-        Robolectric.buildContentProvider(LifecycleRuntimeTrojanProvider.class).create();
     }
 
     @Test

